qemu/tests/balloon_hotplug.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import time
+import random
+
+from virttest.qemu_devices import qdevices
+from virttest import error_context
+from qemu.tests import balloon_check
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test hotplug of balloon devices.
+
+    1) Boot up guest w/o balloon device.
+    2) Hoplug balloon device and check hotplug successfully or not.
+    3) Do memory balloon.
+    4) Unplug balloon device and check unplug successfully or not.
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+    def memory_balloon():
+        if params['os_type'] == 'windows':
+            balloon_test = balloon_check.BallooningTestWin(test, params, env)
+        else:
+            balloon_test = balloon_check.BallooningTestLinux(test, params, env)
+        for tag in params.objects('test_tags'):
+            params_tag = params.object_params(tag)
+            error_context.context("Running %s test" % tag, logging.info)
+            balloon_type = params_tag['balloon_type']
+            min_sz, max_sz = balloon_test.get_memory_boundary(balloon_type)
+            expect_mem = int(random.uniform(min_sz, max_sz))
+        quit_after_test = balloon_test.run_ballooning_test(expect_mem, tag)
+        if quit_after_test:
+            return
+        balloon_test.reset_memory()
+
+    pause = float(params.get("virtio_balloon_pause", 3.0))
+    idx = 0
+    err = ""
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    error_context.context("Hotplug and unplug balloon device in a loop",
+                          logging.info)
+    for i in xrange(int(params.get("virtio_balloon_loops", 3))):
+        new_dev = qdevices.QDevice("virtio-balloon-pci",
```

Comment:
Could you help explain why need add this func()?

    def set_dirty(self):
        """ Increase VM dirtiness (not synchronized with VM) """
        if self.__state >= 0:
            self.__state += 1 
        else:
            self.__state = 1